### PR TITLE
feat(office-365): send documents via email

### PR DESCRIPTION
**Pull request by chillleader**
## Description

This PR adds a new field to our Office 365 connector that allows to send documents in an email.

Since this is an element-template-only connector, we can do this using an intrinsic function `base64`, but for this case it's actually enough to provide a document directly due to how the REST connector works.

Screenshot 2025-03-21 at 14 14 26

## Testing

Follow this guide: https://github.com/camunda/team-connectors/issues/1004
And another guide on how to get a user token: https://github.com/camunda/team-connectors/blob/main/how-to/OUTLOOK_APP_FOR_EMAIL_TESTING.md

## Related issues

closes https://github.com/camunda/team-connectors/issues/1004

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

File: connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json [hunk 1]
```diff
@@ -600,6 +600,24 @@
       },
       "type": "Text"
     },
+    {
+      "id": "users.user.sendMail_attachments",
+      "label": "Attachments",
+      "description": "Attachments. Must be an array of document objects",
```
Comment (johnBgood) [inline connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json:606]: Can we support providing documents either as an array or a single value, as we do in other connectors?

File: connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json [hunk 2]
```diff
@@ -717,7 +735,7 @@
...
+      "value": "={\"message\":{\"subject\":sendMail_subject,\"body\":{\"contentType\":sendMail_body_content_type,\"content\":sendMail_body_content},\"toRecipients\": for currentEmail in sendMail_to_recipients return {\"emailAddress\":{\"address\":currentEmail}},\"ccRecipients\": if sendMail_cc_recipients != null then for currentEmailCC in sendMail_cc_recipients return {\"emailAddress\":{\"address\":currentEmailCC}} else null,\"attachments\": if sendMail_attachments != null then for document in sendMail_attachments return {\"@odata.type\":\"#microsoft.graph.fileAttachment\",\"name\":document.metadata.fileName,\"contentType\":document.metadata.contentType,\"contentBytes\":{\"camunda.function.type\":\"base64\",\"params\":[document]}} else null},\"saveToSentItems\":\"false\"}",
```
Comment (johnBgood) [inline connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json:738]: to be tested, but you shouldn't have to use base64. If the content type is JSON, or no content type is provided (this is our case I believe), then the REST connector will send the content using base64 anyway. 
Again, to be tested :D

**Review by johnBgood — DISMISSED** [review]
Nice! I left a few non-blocking comments

File: connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json (hunk 1 above)
Comment (chillleader) [reply connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json:606]: I don't think we can do this with FEEL, this would require some way to check if the field is a list or a single value. Since we are using a "for" transformation, it requires an array, and all other fields in this connector are required to be arrays. Do you have an idea how we could do this differently?

File: connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json (hunk 2 above)
Comment (johnBgood) [reply connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json:738]: :D

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*

**Review by johnBgood — APPROVED** [review]

**Review by Szik — APPROVED** [review]

File: connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json (hunk 2 above)
Comment (sbuettner) [reply connectors/microsoft/mail/element-templates/microsoft-office365-mail-connector.json]: @johnBgood I checked the docs but it looks like this is not documented anywhere: https://docs.camunda.io/docs/8.7/components/connectors/protocol/rest/#request-body 

Can you add this to the docs?
